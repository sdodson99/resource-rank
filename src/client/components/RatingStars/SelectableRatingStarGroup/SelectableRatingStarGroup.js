import React from 'react';
import PropTypes from 'prop-types';
import styles from './SelectableRatingStarGroup.module.css';
import RatingStarGroup from '../RatingStarGroup/RatingStarGroup';

const SelectableRatingStarGroup = ({
  rating,
  maxRating,
  starSize,
  onRatingChanged,
}) => (
  <div
    className={styles.SelectableRatingStarGroup}
    data-testid="SelectableRatingStarGroup"
  >
    <RatingStarGroup
      rating={rating}
      starSize={starSize}
      maxRating={maxRating}
      onStarClick={(index) => onRatingChanged && onRatingChanged(index + 1)}
    />
  </div>
);

SelectableRatingStarGroup.propTypes = {
  rating: PropTypes.number,
  maxRating: PropTypes.number,
  starSize: PropTypes.number,
  onRatingChanged: PropTypes.func,
};

SelectableRatingStarGroup.defaultProps = {
  rating: 0,
  maxRating: 5,
  starSize: 20,
};

export default SelectableRatingStarGroup;
