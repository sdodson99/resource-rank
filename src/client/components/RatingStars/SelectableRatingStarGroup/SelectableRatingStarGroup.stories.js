import React, { useState } from 'react';
import SelectableRatingStarGroup from './SelectableRatingStarGroup';

export default {
  component: SelectableRatingStarGroup,
  title: 'Components/RatingStars/SelectableRatingStarGroup',
  args: {
    starSize: 50,
  },
};

const Template = (args) => {
  const [rating, setRating] = useState(args.rating ?? 0);

  return (
    <div>
      <SelectableRatingStarGroup
        {...args}
        rating={rating}
        onRatingChanged={setRating}
      />
    </div>
  );
};

export const Default = Template.bind({});
Default.args = {};

export const Size = Template.bind({});
Size.args = {
  starSize: 100,
};

export const MaxRating = Template.bind({});
MaxRating.args = {
  maxRating: 12,
};

export const InitialRating = Template.bind({});
InitialRating.args = {
  rating: 3,
};
