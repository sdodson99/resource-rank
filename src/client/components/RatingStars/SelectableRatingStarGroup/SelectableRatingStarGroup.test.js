import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import SelectableRatingStarGroup from './SelectableRatingStarGroup';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<SelectableRatingStarGroup />', () => {
  it('should mount', () => {
    render(<SelectableRatingStarGroup />);

    const selectableRatingStarGroup = screen.getByTestId(
      'SelectableRatingStarGroup'
    );

    expect(selectableRatingStarGroup).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(
      <SelectableRatingStarGroup rating={3} maxRating={5} starSize={50} />
    );

    expect(component).toMatchSnapshot();
  });

  it('should execute onRatingChanged callback for rating when star clicked', () => {
    const mockOnStarClick = jest.fn();
    render(<SelectableRatingStarGroup onRatingChanged={mockOnStarClick} />);
    const ratingStar = screen.getAllByTestId('RatingStarGroup_RatingStar')[2];

    ratingStar.click();

    expect(mockOnStarClick).toBeCalledWith(3);
  });
});
