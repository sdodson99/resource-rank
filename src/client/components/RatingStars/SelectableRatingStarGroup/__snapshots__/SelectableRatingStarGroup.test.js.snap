// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectableRatingStarGroup /> should render correctly 1`] = `
<div
  className="SelectableRatingStarGroup"
  data-testid="SelectableRatingStarGroup"
>
  <RatingStarGroup
    maxRating={5}
    onStarClick={[Function]}
    rating={3}
    starSize={50}
  />
</div>
`;
